### PR TITLE
support nfs on aws

### PR DESCRIPTION
**Pull request by enderm**

### CODE REVIEW

**Comment by thpang** [comment]
LGTM.

File: roles/baseline/tasks/main.yml [hunk 1]
```diff
@@ -7,10 +7,9 @@
     efsFileSystemId: "{{ V4_CFG_EFS_FSID }}"
     awsRegion: "{{ V4_CFG_EFS_REGION }}"
   when:
-    - PROVIDER is defined
-    - PROVIDER == "aws"
     - V4_CFG_MANAGE_STORAGE is defined
     - V4_CFG_MANAGE_STORAGE
+    - CREATE_EFS_STORAGE_CLASS is defined
```
Comment (NormJohnIV) [inline roles/baseline/tasks/main.yml:12]: Why the new variables CREATE_EFS_STORAGE_CLASS and CREATE_NFS_STORAGE_CLASS? They are unneeded and don't following the naming standard.

File: roles/baseline/tasks/main.yml [hunk 2]
```diff
@@ -33,7 +32,7 @@
     - V4_CFG_NFS_SVR_PATH is defined
     - V4_CFG_MANAGE_STORAGE is defined
     - V4_CFG_MANAGE_STORAGE
-    - PROVIDER != "aws"
+    - CREATE_NFS_STORAGE_CLASS is defined
```
Comment (NormJohnIV) [inline roles/baseline/tasks/main.yml:35]: Why the new variables CREATE_EFS_STORAGE_CLASS and CREATE_NFS_STORAGE_CLASS? They are unneeded and don't following the naming standard.

File: roles/common/tasks/main.yaml [hunk 3]
```diff
@@ -37,9 +37,15 @@
         V4_CFG_NFS_SVR_PATH: "{{ tfstate.rwx_filestore_path.value }}"
       when: 
         - tfstate.rwx_filestore_path is defined
+    - name: tfstate - nfs
```
Comment (NormJohnIV) [inline roles/common/tasks/main.yaml:40]: Why the new variables CREATE_EFS_STORAGE_CLASS and CREATE_NFS_STORAGE_CLASS? They are unneeded and don't following the naming standard.

**Comment by NormJohnIV** [comment]
I am confused by this new variables (CREATE_EFS_STORAGE_CLASS and CREATE_NFS_STORAGE_CLASS). Looking at the logic, you can do the same with the existing variables. 

Ex.
https://github.com/sassoftware/viya4-deployment/blob/156b77a444cdc4bcf3f3f2d4aa77e5ae0b78a90d/roles/baseline/tasks/main.yml#L12

change to 
    - V4_CFG_EFS_FSID is defined
    - V4_CFG_EFS_REGION is defined

**Review by NormJohnIV — CHANGES_REQUESTED** [review]

**Review by NormJohnIV — CHANGES_REQUESTED** [review]
I am confused by this new variables (CREATE_EFS_STORAGE_CLASS and CREATE_NFS_STORAGE_CLASS). Looking at the logic, you can do the same with the existing variables.

Ex.
https://github.com/sassoftware/viya4-deployment/blob/156b77a444cdc4bcf3f3f2d4aa77e5ae0b78a90d/roles/baseline/tasks/main.yml#L12

change to
- V4_CFG_EFS_FSID is defined
- V4_CFG_EFS_REGION is defined

**Comment by enderm** [comment]
it's just a suggestion, to communicate what is needed here. It's your project.
I'll send another commit

**Pull request merged**